core/src/main/scala/io/finch/ToService.scala

```diff
@@ -104,7 +109,7 @@ object ToService {
           req.version,
           includeDateHeader,
           includeServerHeader
-        )).run
+        )).run.onSuccess(withPathContext(underlying))
```

Comment:
What about failure (exception) case?

Reply:
My initial thought was to just leave it empty to signal that it was never set, but that will raise an exception when you attempt to get the field. Maybe this should be the field should be an `Option[String]` or some ADT.

Reply:
Maybe provide it on failure also?